src/main/java/seedu/manager/parser/Parser.java

```diff
@@ -247,6 +255,58 @@ private Command parseRemoveCommand(String input, String[] commandParts) throws I
         }
     }
 
+    /**
+     * Parses the input string to create an Command object based on the provided command parts.
+     * <p>
+     * This method checks the command flag extracted from the command parts. If the command
+     * flag is "-e", it splits the input string to create an EditCommand
+     * for editing an event. If the command flag is "-p", it creates an EditCommand
+     * for editing a participant's details. If neither flag is matched, it throws an InvalidCommandException
+     * with an error message.
+     * </p>
+     *
+     * @param input        the input string containing the command details.
+     * @param commandParts an array of strings representing the parsed command parts,
+     *                     where the second element is the command flag.
+     * @return a Command object representing the parsed command.
+     * @throws InvalidCommandException if the flags are not matched in the command parts.
+     */
+    private Command parseEditCommand(String input, String[] commandParts) throws InvalidCommandException {
+        assert commandParts[0].equalsIgnoreCase(EditCommand.COMMAND_WORD);
+        try {
+            String commandFlag = commandParts[1];
+            String[] inputParts;
+
+            if (commandFlag.equals("-e")) {
+                inputParts = input.split("(-p|-n|-email|-e)");
+                String participantName = inputParts[1].trim();
+                String newNumber = inputParts[2].trim();
+                String newEmail = inputParts[3].trim();
+                String eventName = inputParts[4].trim();
+                return new EditCommand(participantName, newNumber, newEmail, eventName);
+            } else if (commandFlag.equals("-p")) {
```

Comment:
I think you can extract this section (the parsing of the input) out for better SLAP?